packages/framework-provider-aws/src/library/events-searcher-adapter.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  BoosterConfig,
+  EventEnvelope,
+  EventFilter,
+  EventInterface,
+  EventSearchResponse,
+  Logger,
+  UUID,
+} from '@boostercloud/framework-types'
+import { DynamoDB } from 'aws-sdk'
+import { dynamoDbBatchGetLimit, eventsStoreAttributes } from '../constants'
+import { DocumentClient } from 'aws-sdk/clients/dynamodb'
+import { decodeEventStoreSortingKey, partitionKeyForEvent, partitionKeyForIndexByEntity } from './keys-helper'
+
+export async function searchEvents(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  filters: EventFilter
+): Promise<Array<EventSearchResponse>> {
+  logger.debug('Initiating an events search. Filters: ', filters)
+  const timeFilterQuery = buildSearchEventsTimeQuery(filters.from, filters.to)
+  const eventEnvelopes = await executeSearch(dynamoDB, config, logger, filters, timeFilterQuery)
+
+  logger.debug('Events search result: ', eventEnvelopes)
+  return convertToSearchResult(eventEnvelopes)
+}
+
+interface TimeQueryData {
+  expression: string
+  attributeValues: Record<string, string>
+}
+
+function buildSearchEventsTimeQuery(from?: string, to?: string): TimeQueryData {
+  let timeQueryData: TimeQueryData = {
+    expression: '',
+    attributeValues: {},
+  }
+  if (from && to) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} BETWEEN :fromTime AND :toTime`,
+      attributeValues: {
+        ':fromTime': from,
+        ':toTime': to,
+      },
+    }
+  } else if (from) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} >= :fromTime`,
+      attributeValues: { ':fromTime': from },
+    }
+  } else if (to) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} <= :toTime`,
+      attributeValues: { ':toTime': to },
+    }
+  }
+  return timeQueryData
+}
+
+async function executeSearch(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  filters: EventFilter,
+  timeFilterQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  if ('entity' in filters) {
+    if (filters.entityID) {
+      return await searchEventsByEntityAndID(
+        dynamoDB,
+        config,
+        logger,
+        filters.entity,
+        filters.entityID,
+        timeFilterQuery
+      )
+    } else {
+      return await searchEventsByEntity(dynamoDB, config, logger, filters.entity, timeFilterQuery)
+    }
+  } else if ('type' in filters) {
+    return await searchEventsByType(dynamoDB, config, logger, filters.type, timeFilterQuery)
+  } else {
+    throw new Error('Invalid search event query. It is neither an search by "entity" nor a search by "type"')
+  }
+}
+
+async function searchEventsByEntityAndID(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entity: string,
+  entityID: UUID,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: Manage pagination
```

Comment:
TODO: we need to add pagination to all these queries. @NickSeagull is working on a way to do it

Reply:
Check my suggestion